addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -3172,175 +3309,86 @@ export class OdooEditor extends EventTarget {
-                            sel.collapseToEnd();
-                        }
-                    }
-                } else if (splitAroundUrl[i] !== '') {
-                    const textFragments = splitAroundUrl[i].split(/\r?\n/);
-                    let textIndex = 1;
-                    for (const textFragment of textFragments) {
-                        this.execCommand('insertText', textFragment);
-                        if (textIndex < textFragments.length) {
-                            this._applyCommand('oShiftEnter');
-                        }
-                        textIndex++;
+            return;
+        }
+        const insertLink = (url, label) => {
+            this._recordHistorySelection(true);
+            const link = document.createElement('A');
+            link.setAttribute('href', url);
+            for (const [param, value] of Object.entries(this.options.defaultLinkAttributes)) {
+                link.setAttribute(param, `${value}`);
+            }
+            link.innerText = label;
+            if (!sel.isCollapsed) {
+                this.deleteRange(sel);
+            }
+            if (sel.rangeCount) {
+                sel.getRangeAt(0).insertNode(link);
+                sel.collapseToEnd();
+            }
+            this.historyStep();
+        }
+        const text = ev.clipboardData.getData('text/plain');
+        const splitAroundUrl = text.split(URL_REGEX);
+        const selectionIsInsideALink = !!closestElement(sel.anchorNode, 'a');
+
+        if (splitAroundUrl.length === 3 && splitAroundUrl[0] === '' && splitAroundUrl[2] === '') {
+            // Text is a single URL.
+            const url = /^https?:\/\//i.test(text) ? text : 'https://' + text;
+            const youtubeUrl = this.options.allowCommandVideo && YOUTUBE_URL_GET_VIDEO_ID.exec(url);
+            const urlFileExtention = url.split('.').pop();
+            const isImageUrl = ['jpg', 'jpeg', 'png', 'gif'].includes(urlFileExtention.toLowerCase());
+
+            if (!isImageUrl && !youtubeUrl && !selectionIsInsideALink) {
+                insertLink(url, text);
+                return;
+            }
+            const stepIndexBeforeInsert = this._historySteps.length - 1;
+            const insertedNodes = this._applyCommand('insertText', text);
+            const commands = this._getPasteURLCommands(url, text, stepIndexBeforeInsert,
+                { youtubeUrl, isImageUrl, selectionIsInsideALink });
+            if (commands.length === 1) {
+                // Only "Paste as text" with selection inside a link is
+                // available. Text is already inserted.
+                return;
+            }
+            // Command 'insertText' loses cursor position when inside
+            // isolated link, preventing powebox to open
+            if (selectionIsInsideALink && this._fixLinkMutatedElements) {
+                setCursorEnd(peek(insertedNodes));
+            }
```

Comment:
This is a fix for this particular case, but it could be a symptom of a greater issue.
Maybe fix the 'insertText' command to handle isolated links?

Reply:
Well, are you saying that the `insertText` command does not work inside of isolated links ? If that's the case then yes I think it should be fixed in a separate commit as there is no reason for it not to work. Isolated links are just a special case of an editable context inside of a non-editable context. It should correctly write inside the link since it is editable.

Reply:
`insertText` does insert text inside isolated links. But the cursor disappears afterwards.
If the user wants to continue typing, he/she has to click somewhere.
But in case the user pastes an image URL inside an existing link, the missing cursor prevents the powerbox to pop up open.

Reply:
![image](https://user-images.githubusercontent.com/65516503/210221061-8488ba16-4b0a-46eb-b5ea-be47f1d7cb18.png)
I suspect the culprit, in this case, is the line that sets the cursor as `rightPos(currentNode)` in `commands.insert`. This places the cursor in the link's next sibling, which is outside the contenteditable area when the link is isolated.

Reply:
If that is the case then I think `rightPos` probably isn't supposed to exit the editable zone.

Reply:
The decision to drop the "Paste as text" command made things easier here. When pasting an image URL inside an existing link, the only option now is to embed the image. A powerbox with a single command makes no sense, so the image is directly embedded. (please confirm if this is the desired behavior...)
Therefore, no more need for this fix.